src/pip/_internal/utils/_log.py

```diff
@@ -0,0 +1,34 @@
+"""Customize logging
+
+Defines custom logger class for the `logger.verbose(...)` method.
+
+Must be imported before anything else in pip._internals which may call getLogger
+"""
+
+import logging
+from typing import Any, cast
+
+# custom log level for `--verbose` output
+# between DEBUG and INFO
+VERBOSE = 15
+
+
+class VerboseLogger(logging.Logger):
+    """Custom Logger, defining a verbose log-level
+
+    VERBOSE is between INFO and DEBUG.
+    """
+
+    def verbose(self, msg: str, *args: Any, **kwargs: Any) -> None:
+        return self.log(VERBOSE, msg, *args, **kwargs)
+
+
+# setLoggerClass must be called before importing any other pip submodules
+# that call logging.getLogger
+logging.setLoggerClass(VerboseLogger)
+logging.addLevelName(VERBOSE, "VERBOSE")
+
+
+def getLogger(name: str) -> VerboseLogger:
+    """Wraps logging.getLogger in an appropriate type annotation"""
+    return cast(VerboseLogger, logging.getLogger(name))
```

Comment:
I wonder if there is a better way to do this… Maybe supply a custom `.pyi` stub instead?

Reply:
No `.pyi` files please -- this is significantly easier to maintain than separate stub file, and is fairly clear.

Reply:
If you know how to do that, let me know! I spent a couple of hours trying to do just that, and can't figure it out. I find mypy *super* hard to work with.